app/src/main/res/layout/activity_movie_details.xml

```diff
@@ -0,0 +1,150 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.design.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/coordinator_details"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".movie_details.MovieDetailsActivity">
+
+    <android.support.design.widget.AppBarLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="?attr/colorPrimary">
+        <android.support.design.widget.CollapsingToolbarLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            app:contentScrim="?attr/colorPrimary"
+            app:layout_scrollFlags="scroll|exitUntilCollapsed"
+            app:expandedTitleTextAppearance="@style/Theme.Toolbar.Title">
+            <ImageView
+                android:id="@+id/img_context"
+                android:layout_width="match_parent"
+                android:layout_height="200dp"
+                android:scaleType="centerCrop"
+                android:src="@drawable/ic_image_placeholder"
+                app:layout_collapseMode="parallax" />
+            <android.support.v7.widget.Toolbar
+                android:id="@+id/toolbar"
+                android:layout_width="match_parent"
+                android:layout_height="?attr/actionBarSize"
+                app:layout_collapseMode="pin"
+                app:theme="@style/ToolBarStyle"
+                app:titleTextAppearance="@style/Theme.Toolbar.Title"/>
+        </android.support.design.widget.CollapsingToolbarLayout>
+    </android.support.design.widget.AppBarLayout>
+
+    <android.support.v4.widget.NestedScrollView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:clipToPadding="false"
+        app:layout_behavior="@string/appbar_scrolling_view_behavior">
+
+        <LinearLayout
```

Comment:
Have you heard of `ConstraintLayout`? If so, what would be the advantages and disadvantages of using it here instead of `LinearLayout`?

Reply:
Yes, I have used ConstraintLayout. The benefits of it is that it adjusts depending on the screen’s size and it can be added easily with Android Studio. One of the disadvantages comparing with LinearLayout is that it is a bit tricky to position the views in the correct way, because all the views must be connected horizontally and vertically or it generates a layout error. Thinking better about it, the ConstraintLayout would have been a better choice because independently of the child views on the ConstraintLayout, the views would be adjusted according to the size of the screen.